### DISCUSSION TITLE
Regarding What's Api

### DISCUSSION THREAD

**Discussion by vayun-dev** [0426abhishek/whatsapp-business-api#4] [Q&A] [answered]
Please share the what's api list.

**Comment by 0426abhishek** [comment, marked as answer]
> Please share the what's api list.

List of API's used in what's API business:- 

### getTemplatesList
Api to get the list of templates available.
* Type: 'GET'
* Response OK Status: 200

---

### markUserOptIn
Api to mark user opt in for sending the whatsapp message.
* Type: 'GET'
* Content Type: 'application/json'
* Params:
  - mobileNumber: string
* Response OK Status: 200

---

### sendMediaImageMessage
Api to send image along with the caption.
* Type: 'GET'
* Content Type: 'application/json'
* Params:
  - mobileNumber: string
  - imageUrl: string(public available image url)
  - caption: string
* Response OK Status: 200

---

### sendMediaVideoMessage
Api to send video along with the caption.
* Type: 'GET'
* Content Type: 'application/json'
* Params:
  - mobileNumber: string
  - videoUrl: string(public available video url)
  - caption: string
* Response OK Status: 200

---

### sendTextMessage
Api to send text message.
* Type: 'GET'
* Content Type: 'application/json'
* Params:
  - mobileNumber: string
  - message: string(4096 characters limit)
* Response OK Status: 200

---

### sendTemplateMessage
Api to send templated text message.
* Type: 'GET'
* Content Type: 'application/json'
* Params:
  - mobileNumber: string
  - templateId: string
  - templateParams: array of string containing data for each param
  - message: string(4096 characters limit)/object
* Response OK Status: 200
